server/src/modules/clubs/application/use-cases/get-all-clubs/__tests__/get-all-clubs-controller.test.unit.ts

```diff
@@ -0,0 +1,43 @@
+import httpMocks from 'node-mocks-http'
+import { MikroClubRepo } from '../../../../infra/repos/implementations/mikro-club-repo'
+import { DecodedExpressRequest } from '../../../../../../shared/infra/http/routes/decoded-request'
+import { Club } from '../../../../domain/entities/club'
+import { ClubDTO } from '../../../../mappers/club-dto'
+import { GetAllClubsController } from '../get-all-clubs-controller'
+import { createMockClubs } from '../test-utils/create-clubs'
+import { createMockClubDTOs } from '../test-utils/create-club-dtos'
+
+// TODO: how to show developer these mocks are necessary when building a controller? aka must be synced with buildController()
+jest.mock('../../../../infra/repos/implementations/mikro-club-repo')
+
+describe('GetAllClubsController', () => {
+  let mockClubs: Array<Club>
+  let mockClubDTOs: Array<ClubDTO>
+
+  beforeAll(() => {
+    mockClubs = createMockClubs()
+    mockClubDTOs = createMockClubDTOs()
+  })
+
```

Comment:
we should probably add a test when the repo throws instead of resolves the correct promise, and assert that the controller returns a 500. probably wait though until we figure out what we're going to return on our repo methods as per my other comment.

Reply:
if we don't wrap it, wouldn't an unexpected error crash the entire server?